2017/test/LameingoAutonomous.java

```diff
@@ -136,8 +166,29 @@ public void init()
     @Override
     public void start()
     {
-        nearBeaconTask = new DeadReckonTask(this, approachNearBeacon, lightCriteria);
-        addTask(nearBeaconTask);
+        this.addTask(new DeadReckonTask(this, approachNearBeacon, backLightCriteria) {
+            @Override
+            public void handleEvent(RobotEvent e) {
+                if (e instanceof DeadReckonTask.DeadReckonEvent) {
+                    DeadReckonTask.DeadReckonEvent drEvent = (DeadReckonTask.DeadReckonEvent) e;
+
+                    if (drEvent.kind == DeadReckonTask.EventKind.SENSOR_SATISFIED) {
+                        robot.addTask(new DeadReckonTask(robot, lineDetectTurnPath, lightCriteria) {
+                            @Override
+                            public void handleEvent(RobotEvent e)
+                            {
+                                DeadReckonTask.DeadReckonEvent drEvent = (DeadReckonTask.DeadReckonEvent) e;
+
+                                if (drEvent.kind == DeadReckonTask.EventKind.SENSOR_SATISFIED) {
+                                    robot.addTask(new DeadReckonTask(robot, beaconAlignTurn, backLightCriteria));
+                                }
+                            }
```

Comment:
Good to see the use of anonymous classes and event handlers to serialize things.

However, this does quickly get out of hand when you nest 5, 6, or 7 actions.  A preferred approach is to put the next task into it's own function where the function is named appropriately so the reader knows what is happening next.

So here in the if clause of the first sensor satisfied expression you'd call some method like doTurnOnLine() and you'd put the call to robot,addTask() into that method.  This keeps the nesting reasonable and serves to help self-document your software.    Similarly for the second.